components/sock_utils/include/ifaddrs.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include "lwip/sockets.h"
+#include "netdb_macros.h"
+// include also other related headers to simplify porting of linux libs
+#include "getnameinfo.h"
+#include "socketpair.h"
+#include "gai_strerror.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_IDF_TARGET_LINUX
+// namespace with esp_ on linux to avoid duplication of symbols
+#define getifaddrs esp_getifaddrs
+#define freeifaddrs esp_freeifaddrs
+#endif
+
+/**
+ * @brief Simplified version of ifaddr struct
+ */
+struct ifaddrs {
+    struct ifaddrs  *ifa_next;    /* Next item in list */
+    char            *ifa_name;    /* Name of interface */
+    struct sockaddr *ifa_addr;    /* Address of interface */
+    int ifa_flags;
```

Comment:
Should be `unsigned int` per https://man7.org/linux/man-pages/man3/getifaddrs.3.html

Reply:
fixed to use `unsigned int`